### PR TITLE
Fetch selected dataset in remote mode

### LINKED ISSUE #258: Support remote part in data viewer
### Feature you want to implement
I will implement the remote part in data viewer to fetch and show a specific H5 result file.

### How the feature is implemented
For this, the following functions are required.
- Widgets that select date, hour, and RID.
- Function to fetch RID list for a specific date (and hour).
Once a date or an hour is selected, the corresponding RID list is fetched.
- Function to fetch dataset list for a specific RID.
Once a RID is selected, the list of stored dataset names is fetched.  
- Function to fetch a specific dataset.
Once a dataset name is selected, the corresponding dataset is fetched.

To support these features, the corresponding functions should be implemented in ARTIQ-PROXY.

### PR DESCRIPTION
**Pull request by BECATRUE**
This closes #258.

What I have implemented is:
- Fetch the selected dataset in remote mode.
- Handle switching the source mode; realtime and remote.

image

### CODE REVIEW

**Comment by BECATRUE** [comment]
This Pylint warning will be resolved by rebasing `develop` branch after merging #272.
Please wait a moment.

**Comment by BECATRUE** [comment]
Now, there is no Pylint warning! Please check this PR. @kangz12345

File: iquip/apps/dataviewer.py [hunk 1]
```diff
@@ -1129,23 +1205,32 @@ def switchSourceMode(self, buttonId: int):
             self.startRealtimeDatasetListThread()
         else:
             self.realtimeListThread.stop()
-            remotePart: _RemotePart = self.frame.sourceWidget.stack.widget(
-                SourceWidget.ButtonId.REMOTE
-            )
-            self.startRemoteListThread(remotePart.ridComboBox.currentText())
+            self.startRemoteListThread(self.remotePart.ridComboBox.currentText())
+
+    def _handleDatasetClicked(self, name: str):
```
Comment (kangz12345) [inline iquip/apps/dataviewer.py:1211]: Let's annotate as a pyqt slot!

**Review by kangz12345 — CHANGES_REQUESTED** [review]

**Review by kangz12345 — APPROVED** [review]
LGTM

**Pull request merged**